api/api/controllers/search_controller.py

```diff
@@ -191,13 +192,13 @@ def get_excluded_providers_query() -> Q | None:
                 key=FILTERED_PROVIDERS_CACHE_KEY,
                 version=FILTERED_PROVIDERS_CACHE_VERSION,
                 timeout=FILTER_CACHE_TIMEOUT,
-                value=filtered_providers,
+                value=enabled_providers,
             )
         except ConnectionError:
             logger.warning("Redis connect failed, cannot cache filtered providers.")
 
-    if filtered_providers:
-        return Q("terms", provider=filtered_providers)
+    if enabled_providers:
+        return Q("terms", provider=enabled_providers)
```

Comment:
> > All this is to say that I think we can proceed with this PR, and exactly because we rely on ContentProvider to filter sources. I'll review the code again to check that this is true.
> 
> If `ContentProvider` is _only_ for sources then we need to change the filter here to `source`, not `provider`.

This is exactly what I proposed in the first comment of this thread, which it is... deleted?? I don't understand what happened 😳 Anyway, I 100% agree that the current naming is a mess and very confusing between what actually is a source and what is a provider. I'll apply the minimal changes required to fulfill the issue requirement and keep the PR simple. As @obulat suggests, the renaming sure warrants its own issue, maybe even a small project.

> Originally, we only had source. Then, the provider field was added to denote the provider of the metadata about the image [1](https://github.com/WordPress/openverse/pull/4238#user-content-fn-1-12b80993ab3e016fe9624dcd6bf5e692).

That is a curious fact. I thought the `provider` field came before the `source` given that the model's name is `ContentProvider`.

Reply:
If we only care about sources and not providers, then we should "just" (in quotes because maybe it isn't trivial to do) rename ContentProvider to `ContentSource` (although, I don't know why the "content" prefix is necessary) and the variables/serializers/documentation to clarify it (e.g. in the stats endpoint).

We don't need to add structure that is never used, so if `ContentProvider` only ever really means to reference the sources, then we can just forget about representing providers as an exclusive category at the API database level altogether, at least until we actually have a reason to do that.

> This uses the stats endpoint that queries sources, not providers

@obulat the stats endpoint queries `ContentProvider`, so I don't think this is in any way obvious through any of the current implementation, on the name level, certainly not by just reading the API code... and my intuition would be to assume there was a confusion between source and provider, rather than the difference in reference on the frontend to all references in the API were intentional.

Just want to clarify that this _is_ messy and needs to be cleaned up, otherwise we're relying on latent, undocumented understandings of how the usage of that particular model changed, without ever having made the code reflect that, not even at the public documentation level:

https://api.openverse.org/v1/#tag/images/operation/images_stats

The documentation there now says that `display_name` is "The name of content provider, e.g. Flickr", contrasted to `source_name`, which is "The source of the media, e.g. flickr". The difference is unexplained, and there's no way to know that `ContentProvider` actually represents sources based on how the code uses it.

https://github.com/WordPress/openverse/blob/c92d484cd0f356cb71e8fd7fc15d4a45d6523d0f/api/api/views/media_views.py#L206-L217

Note that the serializer, model, and variable name of the retrieved data are all provider, not source.

Really just wanting to make it clear that this _isn't_ clear in the code, and I think we should consider treating resolving that with greater urgency than we have thus far. The source/provider distinction is terribly tedious, and anything we do to clarify it even for ourselves is worth the effort.

Reply:
Thanks for opening the issue, by the way, @krysal! For visibility, it is here:

https://github.com/WordPress/openverse/issues/4346

Reply:
I agree 100% with you @sarayourfriend, so I raised the issue's priority.